javascript/javascript1/week2/smart-ease.io.js

```diff
@@ -0,0 +1,189 @@
+  //weekday names and choose the right index with the index numeration starting from 0;
+  //However .toLocaleDateString seems more elegant way.
+}
+alert(
+  `Your meeting will be on ${getEventWeekday(
+    Number(prompt("Please, enter the number of days left before the meeting"))
+  )}`
+);
+
+//Fourth task code: Weather wear
+/*function shouldWear(temp) {
+  return temp < -60
+    ? "spacesuit"
+    : temp < -30
+    ? "Arctic Expedition & Polar Clothing"
+    : temp < -10
+    ? "Warm jacket and pair of good trousers"
+    : temp < 0
+    ? "Jacket and pants"
+    : temp < 10
+    ? " a parka, biker jacket or leather jacket"
+    : temp < 20
+    ? "Hoodie and jeans"
+    : temp < 30
+    ? "T-shirt and shorts"
+    : temp < 50
+    ? "thawb and kufiyah"
+    : "spacesuite";
+}*/
+//Decided to rewrite conditions using if for better readability:
+
+function shouldWear(temp) {
+  if (temp < -60) {
+    return "spacesuit";
+  } else if (temp < -30) {
+    return "Arctic Expedition & Polar Clothing";
+  } else if (temp < -10) {
+    return "Warm jacket and pair of good trousers";
+  } else if (temp < 0) {
+    return "Jacket and pants";
+  } else if (temp < 10) {
+    return "a parka, biker jacket or leather jacket";
+  } else if (temp < 20) {
+    return "Hoodie and jeans";
+  } else if (temp < 30) {
+    return "T-shirt and shorts";
+  } else if (temp < 50) {
+    return "thawb and kufiyah";
+  } else {
+    return "spacesuite";
+  }
+}
+
+alert(
+  `You probably will be better of wearing ${shouldWear(
+    Number(prompt("Enter current tempreture in C:"))
+  )}`
+);
+
+//Fifth task code: Student manager
+const class23Students = [];
+function addStudentToClass(studentName) {
+  if (class23Students.indexOf(studentName) !== -1) {
+    console.log(`Student ${studentName} is already in class23`);
+  } else if (studentName == "") {
+    console.log("The student's name can't be an empty string!");
+  } else if (class23Students.length < 6 || studentName === "Queen") {
+    class23Students.push(studentName);
+  } else {
+    console.log("Class 23 is full");
+  }
+}
+
+function getNumberOfStudents() {
+  return class23Students.length;
+}
+
+for (let i = 0; i < 4; i++) {
+  addStudentToClass(`Student${i}`);
+}
+addStudentToClass("Student1");
+addStudentToClass("");
+addStudentToClass("Anakin Skywalker");
+addStudentToClass("Darth Wader");
+addStudentToClass("Queen");
+addStudentToClass("Princess Amidala");
+
+console.log(getNumberOfStudents());
+console.log(class23Students);
+
+//Sixth, bonus, task code: Candy helper
+const candyPrice = {
+  Sweet: 0.5,
+  Chocolate: 0.7,
+  Toffee: 1.1,
+  "Chewing-gum": 0.03,
+};
+const boughtCandyPrices = [];
+const amountToSpend = Math.floor(Math.random() * 100);
+console.log(`You can buy candies for ${amountToSpend} credits`);
+
+function addCandy(candyType, weight) {
+  boughtCandyPrices.push(Number((candyPrice[candyType] * weight).toFixed(3)));
```

Comment:
not really necessary but you could store the price in a variable and reuse it again line 152.

Reply:
added a variable. thanks!